io.github.unknownskl.greenlight.yml

```diff
@@ -0,0 +1,76 @@
+app-id: io.github.unknownskl.greenlight
+runtime: org.freedesktop.Platform
+runtime-version: '23.08'
+sdk: org.freedesktop.Sdk
+base: org.electronjs.Electron2.BaseApp
+base-version: '23.08'
+sdk-extensions:
+  - org.freedesktop.Sdk.Extension.node18
+command: run.sh
+separate-locales: false
+finish-args:
+  - --share=ipc
+  - --socket=x11
+  - --socket=pulseaudio
+  - --share=network
+  - --allow=bluetooth
+  - --device=all
+  - --filesystem=/run/udev:ro
+build-options:
+  append-path: /usr/lib/sdk/node18/bin
+  env:
+    NPM_CONFIG_LOGLEVEL: info
+modules:
+  - name: greenlight
+    buildsystem: simple
+    build-options:
+      env:
+        XDG_CACHE_HOME: /run/build/greenlight/flatpak-node/cache
+        npm_config_cache: /run/build/greenlight/flatpak-node/npm-cache
+        npm_config_offline: 'true'
+        yarn_config_cache: /run/build/greenlight/flatpak-node/yarn-mirror
+    build-commands:
+      - cp -a ../extralib -t /app/
+      - yarn --offline
+      - yarn run flatpak-build --offline -- --linux
+      # Add icon, metadata and desktop file
+      - install -Dm 644 ../io.github.unknownskl.greenlight.png ${FLATPAK_DEST}/share/icons/hicolor/512x512/apps/${FLATPAK_ID}.png
+      - install -Dm 644 ../io.github.unknownskl.greenlight.desktop ${FLATPAK_DEST}/share/applications/${FLATPAK_ID}.desktop
+      - install -Dm 644 ../io.github.unknownskl.greenlight.metainfo.xml ${FLATPAK_DEST}/share/metainfo/${FLATPAK_ID}.appdata.xml
+      # Bundle app and dependencies
+      - cp -a dist/linux*unpacked /app/main
+      # Install app wrapper
+      - install -Dm755 -t /app/bin/ ../run.sh
+    subdir: main
+    sources:
+      - type: git
+        url: https://github.com/unknownskl/greenlight
+        tag: v2.0.0-beta15
+        commit: 862b0e2502eeb9539126ba7a8d4aeb630a41b148
+        dest: main
+      - type: git
+        url: https://github.com/Originalimoc/amd-patched-mesa
+        dest: extralib
+        commit: "dd468982c84d77b1679a45f082af102b665dfd26"
+      - yarn-sources.json
+      - type: inline
+        dest-filename: .yarnrc
+        contents: |
+          yarn-offline-mirror "/run/build/greenlight/flatpak-node/yarn-mirror"
+          yarn-offline-mirror-pruning true
+      - type: file
+        path: io.github.unknownskl.greenlight.metainfo.xml
+      - type: file
+        path: io.github.unknownskl.greenlight.png
+      - type: file
+        path: io.github.unknownskl.greenlight.desktop
+      # Wrapper to launch the app
+      - type: script
+        dest-filename: run.sh
+        commands:
+          - export LIBGL_DRIVERS_PATH=/app/extralib/dri:/usr/lib/x86_64-linux-gnu/dri/intel-vaapi-driver:/usr/lib/x86_64-linux-gnu/dri:${LIBGL_DRIVERS_PATH}
+          - export LIBVA_DRIVERS_PATH=/app/extralib/dri:/usr/lib/x86_64-linux-gnu/dri/intel-vaapi-driver:/usr/lib/x86_64-linux-gnu/dri:${LIBVA_DRIVERS_PATH}
```

Comment:
```
3:1214/083616.905801:ERROR:bus.cc(399)] Failed to connect to the bus: Failed to connect to socket /run/dbus/system_bus_socket: No such file or directory
MESA-LOADER: failed to open radeonsi: /usr/lib/x86_64-linux-gnu/dri/radeonsi_dri.so: cannot open shared object file: No such file or directory (search paths /app/extralib/dri:/usr/lib/x86_64-linux-gnu/dri/intel-vaapi-driver:/usr/lib/x86_64-linux-gnu/dri:, suffix _dri)
failed to load driver: radeonsi
MESA-LOADER: failed to open swrast: /usr/lib/x86_64-linux-gnu/dri/swrast_dri.so: cannot open shared object file: No such file or directory (search paths /app/extralib/dri:/usr/lib/x86_64-linux-gnu/dri/intel-vaapi-driver:/usr/lib/x86_64-linux-gnu/dri:, suffix _dri)
failed to load driver: swrast
```

There is nothing in there:

```
flatpak run --command=sh io.github.unknownskl.greenlight
[📦 io.github.unknownskl.greenlight ~]$ ls /usr/lib/x86_64-linux-gnu/dri/
intel-vaapi-driver
```

And this are not set in the runtime.

Reply:
The extension point is GL:

```
[📦 io.github.unknownskl.greenlight ~]$ ls /usr/lib/x86_64-linux-gnu/GL/default/lib/dri/
crocus_dri.so  iris_dri.so	  libgallium_dri.so	   nouveau_dri.so	 r300_dri.so  r600_drv_video.so  radeonsi_drv_video.so	virtio_gpu_dri.so	 vmwgfx_dri.so
i915_dri.so    kms_swrast_dri.so  libgallium_drv_video.so  nouveau_drv_video.so  r600_dri.so  radeonsi_dri.so	 swrast_dri.so		virtio_gpu_drv_video.so  zink_dri.so
[📦 io.github.unknownskl.greenlight ~]$ 
```

Reply:
Also you're hardcoding the path to x86-64, when the flatpak is built for aarch64 too: `$(gcc --print-multiarch)`

Reply:
but I doubt intel-vaapi exist on that platform.

Reply:
Thanks, valid point. Will make some changes and let you know once it has been solved

Reply:
Got the error reproduced on my Steam deck and the new commit should fix the error

Reply:
You forgot to unhardcode the x86_64 paths, the flatpak is built for aarch64 too.